api/views.py

```diff
@@ -51,21 +52,38 @@ def delete(self, request, pk, format=None):
 '''
 
 
+class LectureFilter(FilterSet):
+    name = filters.CharFilter(lookup_expr='icontains')
+    professor = filters.CharFilter(method='find_by_professor')
+
+    class Meta:
+        model = Lecture
+        fields = ['name', 'professor']
+
+    def find_by_professor(self, queryset, professor, value):
+        professor = Professor.objects.get(name__icontains=value)
+        lectures = queryset.filter(professor=professor)
+        return lectures
```

Comment:
해당 필터는 메소드를 추가하지 않고 그냥 filter 선언만으로도 충분할 것 같아보이네용ㅎㅎ
```suggestion
class LectureFilter(FilterSet):
    name = filters.CharFilter(lookup_expr='icontains')
    professor = filters.CharFilter(lookup_expr='icontains')

    class Meta:
        model = Lecture
        fields = ['name', 'professor']
```
  
추가적으로 해당 필터는 메소드 구현이 필요하지는 않아 보이지만, 메소드 구현을 한다면 이런 느낌이 아닐까하네요. 직접 확인해보고 남겨주세요.
```suggestion
class LectureFilter(FilterSet):
    name = filters.CharFilter(lookup_expr='icontains')
    professor = filters.CharFilter(method='find_by_professor')

    class Meta:
        model = Lecture
        fields = ['name', 'professor']
        
    def find_by_professor(self, queryset, name, value):
       return queryset.filter(name__icontains=value)
```

Reply:
저기 있는 professor field가 Professor라는 모델을 참조하는 foreign key라서 저렇게 하면 안되더라구요 ㅜ

Reply:
`professor__name__icontains=value` 이렇게 해도 안되나요?? 한번 해보고 말해주세요~